File: DRM/drm.java
import java.io.*;
public class drm{
  public static void main(String[] args) throws IOException {
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    String linea = in.readLine();
    int longitud = linea.length()/2;
    String mitad1 = linea.substring(0,longitud);
    String mitad2 = linea.substring(longitud);
    int suma1=0,suma2=0;
    for (int i =0;i<longitud ;i++ ) {
      char c= mitad1.charAt(i);
      suma1+=(int)c-65;
      c=mitad2.charAt(i);
      suma2+=(int)c-65;
    }
    suma1%=26;suma2%=26;
    for (int j =0;j<longitud ;j++ ) {
      int cod1=(int)mitad1.charAt(j)+suma1;
      if(cod1>90)cod1-=26;
      int cod2 = (int)mitad2.charAt(j)+suma2;
      if(cod2>90)cod2-=26;
      cod2-=65;
      cod1+=cod2;
      if(cod1>90)cod1-=26;
      System.out.print((char)cod1);
    }
    System.out.println();
  }
}

File: DRM/drm.c
#include <stdio.h>
#include <string.h>
int longitud,suma1,suma2,i;
char linea[15001];
char mitad1[7501];
char mitad2[7501];
int main(int argc, char const *argv[]) {
  scanf("%s",linea );
  longitud=strlen(linea)/2;
  memcpy(mitad1,linea,longitud);
  mitad1[longitud]=0;
  strcpy(mitad2,linea+longitud);
  suma1=suma2=0;

  for (i = 0; i < longitud; i++) {
    suma1+=mitad1[i]-65;
    suma2+=mitad2[i]-65;
  }
  suma1%=26;suma2%=26;
  for (i = 0; i < longitud; i++) {
    int cod1=mitad1[i]+suma1,cod2;
    if (cod1>90) cod1-=26;
    cod2=mitad2[i]+suma2;
    if (cod2>90) cod2-=26;
    cod2-=65;
    cod1+=cod2;
    if (cod1>90) cod1-=26;
    printf("%c",cod1);
  }
  printf("\n");
  return 0;
}
